java-frontend/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -497,4 +497,28 @@ private static void addLearnedConstraint(ImmutableSet.Builder<LearnedConstraint>
     return result.build();
   }
 
+  /**
+   * To be used ONLY by ExplodedGraph viewer
+   */
+  @VisibleForTesting
```

Comment:
This is a rather bold change : adding API on programState for debugging purpose is a show stopper for me.